### PR TITLE
Remove manual hostname verifier support, used to be included only because Java 6

### PR DESCRIPTION
**Pull request by ortibazar**
Refs #21854

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by ktoso** [comment]
LGTM, have you checked we dont refer to this in docs explicitly somewhere still?

**Comment by ortigali** [comment]
I have checked, but i could not find anything about verifySession parameters in docs

**Comment by ktoso** [comment]
Great, thanks for checking - LGTM :)

**Review by ktoso — APPROVED** [review]

File: akka-stream/src/main/scala/akka/stream/scaladsl/TLS.scala [hunk 1]
```diff
@@ -84,6 +84,12 @@ object TLS {
       config.sslEngineConfigurator.configure(engine, sslContext)
       engine.setUseClientMode(role == Client)
 
+      if (!config.config.loose.disableHostnameVerification && engine.getUseClientMode && hostInfo.isDefined) {
+        val parameters = engine.getSSLParameters
+        parameters.setEndpointIdentificationAlgorithm("HTTPS")
```
Comment (jrudolph) [inline akka-stream/src/main/scala/akka/stream/scaladsl/TLS.scala]: This stage is more general than HTTPS and might be used elsewhere.

**Comment by jrudolph** [comment]
Hmm, not so sure this is a good idea.

AFAIK manual hostname verification support is a feature of ssl-config. So, if we want to just ignore some configuration stuff from ssl-config we need to be very explicit about it.

Also, having this hook might be valuable for other reasons. Any kind of custom verification logic could be run here.

**Pull request merged**